flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -1329,10 +1331,35 @@ private void discardCheckpoint(PendingCheckpoint pendingCheckpoint, @Nullable Th
 
 		LOG.info("Discarding checkpoint {} of job {}.", checkpointId, job, cause);
 
-		if (cause == null || cause instanceof CheckpointDeclineException) {
-			pendingCheckpoint.abort(CheckpointFailureReason.CHECKPOINT_DECLINED, cause);
+		if (cause == null) {
```

Comment:
Is there any harm in just rewriting this as
```
		if (cause == null) {
			failPendingCheckpoint(pendingCheckpoint, CheckpointFailureReason.CHECKPOINT_DECLINED);
		} else if (cause instanceof CheckpointException) {
			CheckpointException exception = (CheckpointException) cause;
			failPendingCheckpoint(pendingCheckpoint, exception.getCheckpointFailureReason(), cause);
		} else {
			failPendingCheckpoint(pendingCheckpoint, CheckpointFailureReason.JOB_FAILURE, cause);
		}
```

?

Reply:
If all `CheckpointException` objects come from the Flink code, this style is OK.